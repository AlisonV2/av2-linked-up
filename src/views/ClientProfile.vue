<template>
  <div class="container profile-container">
    <div class="row mb-3">
      <app-title>{{ client.name }}</app-title>
    </div>
    <div class="row">
      <div class="col-12 col-lg-5 img-center">
        <img class="img-fluid" :src="client.thumbnail" :alt="client.name"/>
      </div>
      <div class="col-12 col-lg-7 text-center">
        <p class="accent-text">About</p>
        <p>{{ client.description }}</p>
        <p>
          <span class="accent-text">Location:</span> {{ client.city }} ({{
            client.zip
          }})
        </p>
        <p><span class="accent-text">Tattooed?</span> {{ client.tattooed }}</p>
      </div>
    </div>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports ClientProfile
 * @type {Page}
 * @requires Sentry
 * @vue-computed {object} client - Returned from store
 * @vue-event created - Dispatch store action on created hook
 */
export default {
  name: 'clientProfile',
  data() {
    return {
      client: {
        name: '',
        thumbnail: '',
        description: '',
        city: '',
        zip: '',
        tattooed: '',
      },
    };
  },
  async created() {
    await this.$store
      .dispatch('getClientById', this.$route.params.id)
      .then(() => {
        this.client = this.$store.getters.getClientProfile;
      })
      .catch((err) => Sentry.captureException(err));
  },
};
</script>

<style lang="scss">
.img-fluid {
  max-height: 20rem;
}
</style>
