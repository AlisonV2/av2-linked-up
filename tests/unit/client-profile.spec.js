import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import ClientProfile from '@/views/ClientProfile';
import AppTitle from '@/components/app/Title';
import AppButton from '@/components/app/Button';

const actions = {
  getClientById: jest.fn(),
};

const getters = {
    getClientProfile: jest.fn(),
}

const store = createStore({
  actions,
  getters
});

const $router = {
  push: jest.fn(),
};

const $route = {
  params: { id: '1' },
};

const config = {
  data() {
    return {
      client: {
        id: 1,
        name: 'Client',
        description: 'Description',
        thumbnail: 'Thumbnail',
        city: 'City',
        zip: 'Zip',
        tattooed: 'yes',
      },
    };
  },
  global: {
    plugins: [store],
    stubs: {
      'app-title': AppTitle,
      'app-button': AppButton,
    },
    mocks: {
      $router,
      $route,
    },
  },
};

/**
 * @module ClientProfileTest
 */
describe('ClientProfile.vue', () => {
  it('getClientById should be called', async () => {
    const component = shallowMount(ClientProfile, config);
    ClientProfile.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getClientById).toHaveBeenCalled;
  });
  it('Computed client should call getters', async () => {
    const component = shallowMount(ClientProfile, config);
    ClientProfile.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getClientProfile).toHaveBeenCalled;
  });
});
