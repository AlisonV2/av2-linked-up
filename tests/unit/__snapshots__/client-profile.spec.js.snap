// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientProfile.vue ClientProfile Snapshot 1`] = `
<div
  class="container profile-container"
>
  <div
    class="row mb-3"
  >
    <div
      class="col-12 text-center mb-4 app-title"
    >
      <h2
        class="title"
      >
        
        Client
        
      </h2>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-12 col-lg-5 img-center"
    >
      <img
        alt="Client"
        class="img-fluid"
        src="Thumbnail"
      />
    </div>
    <div
      class="col-12 col-lg-7 text-center"
    >
      <p
        class="accent-text"
      >
        About
      </p>
      <p>
        Description
      </p>
      <p>
        <span
          class="accent-text"
        >
          Location:
        </span>
         City (Zip) 
      </p>
      <p>
        <span
          class="accent-text"
        >
          Tattooed?
        </span>
         yes
      </p>
    </div>
  </div>
</div>
`;
